### PR TITLE
Shell command for screed functions and simplified API for DB creation

### PR DESCRIPTION
**Pull request by luizirber**
Related to dib-lab/khmer#1482

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality, is it tested?
- [x] `make format diff_pylint_report doc` Is it well formatted?
- [x] Is it documented in the `ChangeLog`?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

Add a `screed` command, with options to:
- create a screed DB: `createdb`
- dump a DB to FASTA or FASTQ: `dump_to_fasta`, `dump_to_fastq`

### CODE REVIEW

**Comment by ctb** [comment]
Suggest 'python -m screed.db'

**Comment by luizirber** [comment]
rename `createdb` to `db`, and remove `fadbm` and `fqdbm` (because they are redundant)?

**Comment by ctb** [comment]
yes!

**Comment by luizirber** [comment]
Done. Removed `screed fadbm` and `screed fqdbm`, and replaced `screed.fadbm` and `screed.fqdbm` implementation with `screed.createscreed.main`.

**Comment by luizirber** [comment]
Pending changelog, but otherwise good to go! @ctb @standage

**Comment by ctb** [comment]
Changelog format update now merged in; can you update?

**Comment by luizirber** [comment]
@ctb updated

**Comment by ctb** [comment]
Suggest a few additions, I'm happy to implement -- let me know.

- [ ] by default dump to stdout (or, allow '-' to specify stdout)
- [ ] print the number of records dumped to stderr

**Comment by ctb** [comment]
Suggest eliminating fadbm.py and fqdbm.py, since they're basically empty now.

**Comment by luizirber** [comment]
@ctb go for it

**Comment by ctb** [comment]
OK, I'll take this on.

**Comment by ctb** [comment]
Ready for review @standage

File: screed/__main__.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self):
+        parser = argparse.ArgumentParser(
+            description="",
+            usage='''screed <command> [<args>]
+
+Commands can be:
+
+    db <filename>                 "Creates a screed database."
+    dump_to_fasta <db> <output>   "Convert a screed database to a FASTA file"
+    dump_to_fasta <db> <output>   "Convert a screed database to a FASTQ file"
```
Comment (standage) [inline screed/__main__.py]: Copy/paste error: second line should be `dump_to_fastq`.

But on that note, what about something less verbose: `dump_fastq` or even `fastq`? Would that be too ambiguous?

Comment (standage) [inline screed/__main__.py]: I'd be happy to take care of this, but any comments on `dump_to_fastq` vs just `fastq`?

**Comment by ctb** [comment]
I like removing the _to_

**Comment by standage** [comment]
Done. Now we have no unbiased third party to merge :-)

**Comment by luizirber** [comment]
Help us @camillescott,

File: setup.py [hunk 2]
```diff
@@ -31,5 +31,8 @@
       setup_requires=['pytest-runner'],
       tests_require=['pytest >= 3.0', 'pytest-cov'],
       cmdclass=versioneer.get_cmdclass(),
-      install_requires=['bz2file']
-      )
+      install_requires=['bz2file'],
+      entry_points={'console_scripts': [
+          'screed = screed.__main__:main'
+          ]
+      })
```
Comment (standage) [inline setup.py]: This "entry points" functionality is pretty cool!

File: screed/__main__.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    def __init__(self):
+        parser = argparse.ArgumentParser(
+            description="",
+            usage='''screed <command> [<args>]
+
+Available:
+
+    db <filename>               Creates a screed database.
+    dump_fasta <db> <output>    Convert a screed database to a FASTA file
+    dump_fastq <db> <output>    Convert a screed database to a FASTQ file
```
Comment (standage) [inline screed/__main__.py]: I still think `fastq` and `fasta` are simpler and more concise, but for now this is fine.

**Pull request merged**